diff -ru Team/amulet_map_editor/api/framework/pages/main_menu.py Ryan/amulet_map_editor/api/framework/pages/main_menu.py
--- Team/amulet_map_editor/api/framework/pages/main_menu.py	2025-07-25 18:34:16.671047098 -0400
+++ Ryan/amulet_map_editor/api/framework/pages/main_menu.py	2025-07-25 23:47:27.402811394 -0400
@@ -36,12 +36,12 @@
         name_sizer.Add(icon, flag=wx.CENTER)
 
         self._amulet_name = wx.StaticText(self)
-        self._amulet_name.SetFont(wx.Font(40, wx.DEFAULT, wx.NORMAL, wx.NORMAL))
+        self._amulet_name.SetFont(wx.Font(40, wx.DECORATIVE, wx.NORMAL, wx.NORMAL))
         name_sizer.Add(
             self._amulet_name, flag=wx.CENTER | wx.LEFT | wx.RIGHT, border=10
         )
         name_sizer.Add(icon2, flag=wx.CENTER)
-        button_font = wx.Font(20, wx.DEFAULT, wx.NORMAL, wx.NORMAL)
+        button_font = wx.Font(20, wx.DECORATIVE, wx.NORMAL, wx.NORMAL)
         self._open_world_button = wx.Button(self, size=(400, 70))
         self._open_world_button.SetFont(button_font)
         self._open_world_button.Bind(
@@ -92,9 +92,6 @@
                 "https://raw.githubusercontent.com/Amulet-Team/sponsors/main/sponsors.json"
             ) as f:
                 sponsors = json.load(f)
-        except Exception:
-            pass
-        else:
             github_sponsor_text = wx.TextCtrl(
                 self,
                 value="   ".join(sponsors),
@@ -103,6 +100,8 @@
             )
             github_sponsor_text.SetBackgroundColour(self.GetBackgroundColour())
             sponsor_sizer.Add(github_sponsor_text, 1)
+        except (URLError, json.JSONDecodeError):
+            pass
 
         self._load_strings()
 
Only in Ryan/amulet_map_editor/api/framework/pages: main_menu.py.rej
diff -ru Team/amulet_map_editor/api/opengl/canvas/canvas.py Ryan/amulet_map_editor/api/opengl/canvas/canvas.py
--- Team/amulet_map_editor/api/opengl/canvas/canvas.py	2025-07-25 18:34:17.851086830 -0400
+++ Ryan/amulet_map_editor/api/opengl/canvas/canvas.py	2025-07-25 23:47:27.462813324 -0400
@@ -1,6 +1,7 @@
 from typing import Optional, Callable
 import logging
 
+import sys
 import wx
 from wx.glcanvas import GLCanvas, GLAttributes, GLContext, GLContextAttrs
 from OpenGL.GL import (
@@ -28,8 +29,12 @@
 Objects that need to bind textures or data should do so in the draw function so they can be sure the context is set.
 """
 
+if sys.platform == "linux":
+    Canvas_Type = wx.Window
+else:
+    Canvas_Type = GLCanvas
 
-class BaseCanvas(GLCanvas):
+class BaseCanvas(Canvas_Type):
     _context: Optional[GLContext]
 
     def __init__(self, parent: wx.Window):
@@ -42,12 +47,38 @@
         display_attributes.PlatformDefaults().MinRGBA(8, 8, 8, 8).DoubleBuffer().Depth(
             24
         ).EndList()
-        super().__init__(
-            parent,
-            display_attributes,
-            size=parent.GetClientSize(),
-            style=wx.WANTS_CHARS,
-        )
+        if Canvas_Type == wx.Window:
+            super().__init__(parent)
+            # self.SetSizer(self._canvas_sizer)
+            self._opengl_canvas = GLCanvas( self,
+                display_attributes,
+                size=parent.GetClientSize(),
+                style=wx.WANTS_CHARS,
+            )
+
+            def forward_event(event):
+                # Create a new mouse event and send it to the parent
+                new_event = wx.MouseEvent(event.GetEventType())
+                new_event.SetPosition(event.GetPosition())
+                wx.PostEvent(self, new_event)
+                event.Skip()  # Continue processing normally
+            self._opengl_canvas.Bind(wx.EVT_RIGHT_DOWN, forward_event)
+            self._opengl_canvas.Bind(wx.EVT_RIGHT_UP, forward_event)
+            self._opengl_canvas.Bind(wx.EVT_LEFT_DOWN, forward_event)
+            self._opengl_canvas.Bind(wx.EVT_LEFT_UP, forward_event)
+            self._opengl_canvas.Bind(wx.EVT_MOTION, forward_event)
+            self._opengl_canvas.Bind(wx.EVT_MOUSEWHEEL, forward_event)
+            self._opengl_canvas.Bind(wx.EVT_SIZING, self.resize)
+        elif Canvas_Type == GLCanvas:
+            super().__init__(
+                parent,
+                display_attributes,
+                size=parent.GetClientSize(),
+                style=wx.WANTS_CHARS,
+            )
+            self._opengl_canvas = self
+        else:
+            raise NotImplementedError
 
         # Amulet-Team/Amulet-Map-Editor#84
         # Amulet-Team/Amulet-Map-Editor#597
@@ -58,7 +89,7 @@
             ctx_attrs.OGLVersion(3, 3)
             ctx_attrs.CoreProfile()
             ctx_attrs.EndList()
-            ctx = GLContext(self, ctxAttrs=ctx_attrs)
+            ctx = GLContext(self._opengl_canvas, ctxAttrs=ctx_attrs)
             if ctx.IsOK():
                 return ctx
             return None
@@ -69,7 +100,7 @@
             ctx_attrs.OGLVersion(2, 1)
             ctx_attrs.CompatibilityProfile()
             ctx_attrs.EndList()
-            ctx = GLContext(self, ctxAttrs=ctx_attrs)
+            ctx = GLContext(self._opengl_canvas, ctxAttrs=ctx_attrs)
             if ctx.IsOK() and glInitExplicitAttribLocationARB():
                 return ctx
             return None
@@ -84,6 +115,10 @@
 
         self.Bind(wx.EVT_SHOW, self._on_show)
 
+    def resize(self, event):
+        self._opengl_canvas.SetCurrent(self._context)
+        glViewport(0, 0, event.GetSize().x, event.GetSize().y)
+
     @property
     def context(self) -> GLContext:
         return self._context
@@ -101,7 +136,7 @@
 
     def _init_opengl(self):
         """Set up the OpenGL state after the window is first shown."""
-        self.SetCurrent(self._context)
+        self._opengl_canvas.SetCurrent(self._context)
         gl_version = glGetString(GL_VERSION)
         if isinstance(gl_version, bytes):
             gl_version = gl_version.decode("utf-8")
diff -ru Team/amulet_map_editor/api/opengl/textureatlas.py Ryan/amulet_map_editor/api/opengl/textureatlas.py
--- Team/amulet_map_editor/api/opengl/textureatlas.py	2025-07-25 18:34:17.981091207 -0400
+++ Ryan/amulet_map_editor/api/opengl/textureatlas.py	2025-07-25 23:47:27.462813324 -0400
@@ -256,7 +256,7 @@
 
 
 def create_atlas(
-    texture_tuple: Tuple[str, ...],
+    texture_tuple: Tuple[str, ...]
 ) -> Tuple[Image.Image, Dict[str, Tuple[float, float, float, float]]]:
     atlas_iter = create_atlas_iter(texture_tuple)
     try:
diff -ru Team/amulet_map_editor/api/wx/ui/nbt_editor.py Ryan/amulet_map_editor/api/wx/ui/nbt_editor.py
--- Team/amulet_map_editor/api/wx/ui/nbt_editor.py	2025-07-25 18:34:18.031092891 -0400
+++ Ryan/amulet_map_editor/api/wx/ui/nbt_editor.py	2025-07-25 23:47:27.472813646 -0400
@@ -13,7 +13,7 @@
 
 nbt_resources = image.nbt
 
-NBT_FILE = b"\x0a\x00\x0b\x68\x65\x6c\x6c\x6f\x20\x77\x6f\x72\x6c\x64\x08\x00\x04\x6e\x61\x6d\x65\x00\x09\x42\x61\x6e\x61\x6e\x72\x61\x6d\x61\x00"
+NBT_FILE = b"\x0A\x00\x0B\x68\x65\x6C\x6C\x6F\x20\x77\x6F\x72\x6C\x64\x08\x00\x04\x6E\x61\x6D\x65\x00\x09\x42\x61\x6E\x61\x6E\x72\x61\x6D\x61\x00"
 
 
 class NBTRadioButton(simple.SimplePanel):
diff -ru Team/amulet_map_editor/api/wx/util/key_config.py Ryan/amulet_map_editor/api/wx/util/key_config.py
--- Team/amulet_map_editor/api/wx/util/key_config.py	2025-07-25 18:34:18.051093564 -0400
+++ Ryan/amulet_map_editor/api/wx/util/key_config.py	2025-07-25 23:47:27.482813967 -0400
@@ -271,7 +271,7 @@
 
 
 def serialise_key_event(
-    evt: Union[wx.KeyEvent, wx.MouseEvent],
+    evt: Union[wx.KeyEvent, wx.MouseEvent]
 ) -> Optional[SerialisedKeyType]:
     if isinstance(evt, wx.KeyEvent):
         key = evt.GetUnicodeKey() or evt.GetKeyCode()
diff -ru Team/amulet_map_editor/lang/en.lang Ryan/amulet_map_editor/lang/en.lang
--- Team/amulet_map_editor/lang/en.lang	2025-07-25 18:34:18.081094574 -0400
+++ Ryan/amulet_map_editor/lang/en.lang	2025-07-25 23:47:27.492814289 -0400
@@ -265,7 +265,7 @@
 program_3d_edit.file_ui.version_tooltip=Platform and data version of the world
 program_3d_edit.file_ui.projection_tooltip=Change view
 program_3d_edit.file_ui.location_tooltip=Move camera
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=Camera speed in blocks per second
 program_3d_edit.file_ui.speed_dialog_name=Set camera speed
 program_3d_edit.file_ui.dim_tooltip=Select dimension
diff -ru Team/amulet_map_editor/lang/es_MX.lang Ryan/amulet_map_editor/lang/es_MX.lang
--- Team/amulet_map_editor/lang/es_MX.lang	2025-07-25 18:34:18.091094911 -0400
+++ Ryan/amulet_map_editor/lang/es_MX.lang	2025-07-25 23:47:27.502814611 -0400
@@ -259,7 +259,7 @@
 program_3d_edit.file_ui.version_tooltip=Plataforma y versión de datos del mundo
 program_3d_edit.file_ui.projection_tooltip=Cambiar vista
 program_3d_edit.file_ui.location_tooltip=Mover cámara
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=Velocidad de la cámara en bloques por segundo
 program_3d_edit.file_ui.speed_dialog_name=Asignar velocidad de cámara
 program_3d_edit.file_ui.dim_tooltip=Seleccionar Dimensión
diff -ru Team/amulet_map_editor/lang/id.lang Ryan/amulet_map_editor/lang/id.lang
--- Team/amulet_map_editor/lang/id.lang	2025-07-25 18:34:18.101095248 -0400
+++ Ryan/amulet_map_editor/lang/id.lang	2025-07-25 23:47:27.502814611 -0400
@@ -222,7 +222,7 @@
 program_3d_edit.file_ui.version_tooltip=Platform dan versi data dunia
 program_3d_edit.file_ui.projection_tooltip=Ubah tampilan
 program_3d_edit.file_ui.location_tooltip=Geser Kamera
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=Kecepatan kamera dalam blok per detik
 program_3d_edit.file_ui.speed_dialog_name=Atur kecepatan kamera
 program_3d_edit.file_ui.dim_tooltip=Pilih dimensi
diff -ru Team/amulet_map_editor/lang/ja.lang Ryan/amulet_map_editor/lang/ja.lang
--- Team/amulet_map_editor/lang/ja.lang	2025-07-25 18:34:18.111095584 -0400
+++ Ryan/amulet_map_editor/lang/ja.lang	2025-07-25 23:47:27.512814932 -0400
@@ -259,7 +259,7 @@
 program_3d_edit.file_ui.version_tooltip=このワールドのプラットフォームとバージョン
 program_3d_edit.file_ui.projection_tooltip=視点の切替
 program_3d_edit.file_ui.location_tooltip=カメラを移動
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=カメラ速度（ブロック/秒)
 program_3d_edit.file_ui.speed_dialog_name=カメラスピードの設定
 program_3d_edit.file_ui.dim_tooltip=ディメンションを選択
diff -ru Team/amulet_map_editor/lang/ko.lang Ryan/amulet_map_editor/lang/ko.lang
--- Team/amulet_map_editor/lang/ko.lang	2025-07-25 18:34:18.111095584 -0400
+++ Ryan/amulet_map_editor/lang/ko.lang	2025-07-25 23:47:27.522815254 -0400
@@ -259,7 +259,7 @@
 program_3d_edit.file_ui.version_tooltip=세계의 플랫폼 및 데이터 버전
 program_3d_edit.file_ui.projection_tooltip=보기 변경
 program_3d_edit.file_ui.location_tooltip=카메라 이동
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=초당 블록 이동 속도
 program_3d_edit.file_ui.speed_dialog_name=카메라 속도 설정
 program_3d_edit.file_ui.dim_tooltip=차원 선택
diff -ru Team/amulet_map_editor/lang/pt_BR.lang Ryan/amulet_map_editor/lang/pt_BR.lang
--- Team/amulet_map_editor/lang/pt_BR.lang	2025-07-25 18:34:18.121095921 -0400
+++ Ryan/amulet_map_editor/lang/pt_BR.lang	2025-07-25 23:47:27.532815576 -0400
@@ -259,7 +259,7 @@
 program_3d_edit.file_ui.version_tooltip=Plataforma e versão de dados do mundo
 program_3d_edit.file_ui.projection_tooltip=Alterar visualização
 program_3d_edit.file_ui.location_tooltip=Mover Câmera
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=Velocidade da câmera em blocos por segundo
 program_3d_edit.file_ui.speed_dialog_name=Definir velocidade da câmera
 program_3d_edit.file_ui.dim_tooltip=Selecione a dimensão
diff -ru Team/amulet_map_editor/lang/vi.lang Ryan/amulet_map_editor/lang/vi.lang
--- Team/amulet_map_editor/lang/vi.lang	2025-07-25 18:34:18.141096594 -0400
+++ Ryan/amulet_map_editor/lang/vi.lang	2025-07-25 23:47:27.542815897 -0400
@@ -265,7 +265,7 @@
 program_3d_edit.file_ui.version_tooltip=Phiên bản dữ liệu và nền tảng của thế giới
 program_3d_edit.file_ui.projection_tooltip=Thay đổi góc nhìn
 program_3d_edit.file_ui.location_tooltip=Di chuyển máy quay
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=Tốc độ máy quay theo khối trên giây
 program_3d_edit.file_ui.speed_dialog_name=Đặt tốc độ máy quay
 program_3d_edit.file_ui.dim_tooltip=Lựa chọn chiều không gian
diff -ru Team/amulet_map_editor/lang/zh_CN.lang Ryan/amulet_map_editor/lang/zh_CN.lang
--- Team/amulet_map_editor/lang/zh_CN.lang	2025-07-25 18:34:18.141096594 -0400
+++ Ryan/amulet_map_editor/lang/zh_CN.lang	2025-07-25 23:47:27.542815897 -0400
@@ -305,7 +305,7 @@
 program_3d_edit.file_ui.version_tooltip=世界的平台和数据版本
 program_3d_edit.file_ui.projection_tooltip=改变视图
 program_3d_edit.file_ui.location_tooltip=移动相机
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=相机速度（每秒移动的方块数）
 program_3d_edit.file_ui.speed_dialog_name=设置相机速度
 program_3d_edit.file_ui.dim_tooltip=选择维度
diff -ru Team/amulet_map_editor/lang/zh_TW.lang Ryan/amulet_map_editor/lang/zh_TW.lang
--- Team/amulet_map_editor/lang/zh_TW.lang	2025-07-25 18:34:18.151096931 -0400
+++ Ryan/amulet_map_editor/lang/zh_TW.lang	2025-07-25 23:47:27.552816219 -0400
@@ -305,7 +305,7 @@
 program_3d_edit.file_ui.version_tooltip=世界的平臺和數據版本
 program_3d_edit.file_ui.projection_tooltip=改變視圖
 program_3d_edit.file_ui.location_tooltip=移動相機
-program_3d_edit.file_ui.speed_blocks_per_second=b/s
+program_3d_edit.file_ui.speed_blocks_per_second=bps
 program_3d_edit.file_ui.speed_tooltip=相機速度（每秒移動的方塊數）
 program_3d_edit.file_ui.speed_dialog_name=設置相機速度
 program_3d_edit.file_ui.dim_tooltip=選擇維度
diff -ru Team/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py Ryan/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py
--- Team/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py	2025-07-25 18:34:18.321102655 -0400
+++ Ryan/amulet_map_editor/programs/edit/api/canvas/base_edit_canvas.py	2025-07-25 23:47:27.562816540 -0400
@@ -250,7 +250,7 @@
 
     def enable(self):
         """Enable the canvas and start it working."""
-        self.SetCurrent(self._context)
+        self._opengl_canvas.SetCurrent(self._context)
         self.renderer.enable()
         self.buttons.enable()
 
@@ -312,9 +312,9 @@
         evt.Skip()
 
     def _set_size(self):
-        size = self.GetClientSize() * self.GetContentScaleFactor()
+        size = self._opengl_canvas.GetClientSize() * self._opengl_canvas.GetContentScaleFactor()
         width, height = size
-        self.SetCurrent(self._context)
+        self._opengl_canvas.SetCurrent(self._context)
         glViewport(0, 0, width, height)
         if height > 0:
             self.camera.aspect_ratio = width / height
diff -ru Team/amulet_map_editor/programs/edit/api/canvas/edit_canvas.py Ryan/amulet_map_editor/programs/edit/api/canvas/edit_canvas.py
--- Team/amulet_map_editor/programs/edit/api/canvas/edit_canvas.py	2025-07-25 18:34:18.331102992 -0400
+++ Ryan/amulet_map_editor/programs/edit/api/canvas/edit_canvas.py	2025-07-25 23:47:27.572816862 -0400
@@ -1,6 +1,7 @@
 import logging
 import warnings
 import wx
+import sys
 from typing import Callable, TYPE_CHECKING, Any, Generator, Optional
 from types import GeneratorType
 from threading import RLock, Thread
@@ -163,8 +164,18 @@
 
     def _init_opengl(self):
         super()._init_opengl()
-        self._file_panel = FilePanel(self)
-        self._canvas_sizer.Add(self._file_panel, 0, wx.EXPAND, 0)
+        if sys.platform == "linux":
+            self.panel = wx.Panel(self, style=wx.STAY_ON_TOP)
+            self.button_sizer = wx.BoxSizer(wx.HORIZONTAL)
+            self.button_sizer.AddStretchSpacer(1)
+            self.button_sizer.Add(self.panel, 0, wx.EXPAND, 0)
+            self._file_panel = FilePanel(self, self.panel)
+            self.panel.SetSizer(self._file_panel)
+            self.panel.SetBackgroundColour((155, 178, 216, 255))
+            self._canvas_sizer.Add(self.button_sizer, 0, wx.EXPAND, 0)
+        else:
+            self._file_panel = FilePanel(self, self)
+            self._canvas_sizer.Add(self._file_panel, 0, wx.EXPAND, 0)
         self._tool_sizer = ToolManagerSizer(self)
         self._canvas_sizer.Add(self._tool_sizer, 1, wx.EXPAND, 0)
 
diff -ru Team/amulet_map_editor/programs/edit/api/renderer.py Ryan/amulet_map_editor/programs/edit/api/renderer.py
--- Team/amulet_map_editor/programs/edit/api/renderer.py	2025-07-25 18:34:18.391105012 -0400
+++ Ryan/amulet_map_editor/programs/edit/api/renderer.py	2025-07-25 23:47:27.582817184 -0400
@@ -235,13 +235,13 @@
 
         def end_draw(self):
             """Run commands after drawing."""
-            self.canvas.SwapBuffers()
+            self.canvas._opengl_canvas.SwapBuffers()
 
     else:
 
         def end_draw(self):
             """Run commands after drawing."""
-            self.canvas.SwapBuffers()
+            self.canvas._opengl_canvas.SwapBuffers()
             self._chunk_generator.thread_action()
 
     def _gc(self, event):
diff -ru Team/amulet_map_editor/programs/edit/api/ui/file.py Ryan/amulet_map_editor/programs/edit/api/ui/file.py
--- Team/amulet_map_editor/programs/edit/api/ui/file.py	2025-07-25 18:34:18.401105348 -0400
+++ Ryan/amulet_map_editor/programs/edit/api/ui/file.py	2025-07-25 23:47:27.582817184 -0400
@@ -1,5 +1,4 @@
 from typing import TYPE_CHECKING, Optional
-from math import floor, log10
 import wx
 
 from amulet_map_editor.programs.edit.api.edit_canvas_container import (
@@ -25,21 +24,14 @@
     from amulet_map_editor.programs.edit.api.canvas import EditCanvas
 
 
-def _format_float(num: float) -> str:
-    if num < 100:
-        return f"{num:.0{max(0, 2 - floor(log10(num)))}f}".rstrip("0").rstrip(".")
-    else:
-        return f"{num:.0f}"
-
-
 class FilePanel(wx.BoxSizer, EditCanvasContainer):
-    def __init__(self, canvas: "EditCanvas"):
+    def __init__(self, canvas: "EditCanvas", wx_parent):
         wx.BoxSizer.__init__(self, wx.HORIZONTAL)
         EditCanvasContainer.__init__(self, canvas)
 
         level = self.canvas.world
         self._version_text = wx.StaticText(
-            canvas,
+            wx_parent,
             label=f"{level.level_wrapper.platform}, {level.level_wrapper.version}",
         )
         self._version_text.SetToolTip(
@@ -47,7 +39,7 @@
         )
         self.Add(self._version_text, 0)
         self.AddStretchSpacer(1)
-        self._projection_button = wx.Button(canvas, label="3D")
+        self._projection_button = wx.Button(wx_parent, label="3D")
         self._projection_button.SetToolTip(
             lang.get("program_3d_edit.file_ui.projection_tooltip")
         )
@@ -56,7 +48,7 @@
             self._projection_button, 0, wx.TOP | wx.BOTTOM | wx.RIGHT | wx.CENTER, 5
         )
         self._location_button = wx.Button(
-            canvas, label=", ".join([f"{s:.2f}" for s in self.canvas.camera.location])
+            wx_parent, label=", ".join([f"{s:.2f}" for s in self.canvas.camera.location])
         )
         self._location_button.SetToolTip(
             lang.get("program_3d_edit.file_ui.location_tooltip")
@@ -66,20 +58,20 @@
 
         def set_speed(evt):
             dialog = SpeedSelectDialog(
-                canvas, self.canvas.camera.move_speed * 1000 / 33
+                wx_parent, self.canvas.camera.move_speed * 1000 / 33
             )
             if dialog.ShowModal() == wx.ID_OK:
                 self.canvas.camera.move_speed = dialog.speed * 33 / 1000
 
         self._speed_button = wx.Button(
-            canvas,
-            label=f"{_format_float(self.canvas.camera.move_speed * 1000 / 33)} {lang.get('program_3d_edit.file_ui.speed_blocks_per_second')}",
+            wx_parent,
+            label=f"{self.canvas.camera.move_speed*1000/33:.4g}{lang.get('program_3d_edit.file_ui.speed_blocks_per_second')}",
         )
         self._speed_button.SetToolTip(lang.get("program_3d_edit.file_ui.speed_tooltip"))
         self._speed_button.Bind(wx.EVT_BUTTON, set_speed)
         self.Add(self._speed_button, 0, wx.TOP | wx.BOTTOM | wx.RIGHT | wx.CENTER, 5)
 
-        self._dim_options = SimpleChoiceAny(canvas)
+        self._dim_options = SimpleChoiceAny(wx_parent)
         self._dim_options.SetToolTip(lang.get("program_3d_edit.file_ui.dim_tooltip"))
         self._dim_options.SetItems(level.level_wrapper.dimensions)
         self._dim_options.SetValue(level.level_wrapper.dimensions[0])
@@ -88,7 +80,7 @@
         self.Add(self._dim_options, 0, wx.TOP | wx.BOTTOM | wx.RIGHT | wx.CENTER, 5)
 
         def create_button(text, operation):
-            button = wx.Button(canvas, label=text)
+            button = wx.Button(wx_parent, label=text)
             button.Bind(wx.EVT_BUTTON, operation)
             self.Add(button, 0, wx.TOP | wx.BOTTOM | wx.RIGHT, 5)
             return button
@@ -114,7 +106,7 @@
         self._save_button.SetToolTip(lang.get("program_3d_edit.file_ui.save_tooltip"))
 
         close_button = wx.BitmapButton(
-            canvas, bitmap=image.icon.tablericons.square_x.bitmap(20, 20)
+            wx_parent, bitmap=image.icon.tablericons.square_x.bitmap(20, 20)
         )
         close_button.SetToolTip(lang.get("program_3d_edit.file_ui.close_tooltip"))
         close_button.Bind(
@@ -187,7 +179,7 @@
         evt.Skip()
 
     def _on_speed_change(self, evt):
-        label = f"{_format_float(self.canvas.camera.move_speed * 1000 / 33)} {lang.get('program_3d_edit.file_ui.speed_blocks_per_second')}"
+        label = f"{self.canvas.camera.move_speed*1000/33:.4g}{lang.get('program_3d_edit.file_ui.speed_blocks_per_second')}"
         old_label = self._speed_button.GetLabel()
         self._speed_button.SetLabel(label)
         if len(label) != len(old_label):
diff -ru Team/amulet_map_editor/programs/edit/api/ui/tool/base_operation_choice.py Ryan/amulet_map_editor/programs/edit/api/ui/tool/base_operation_choice.py
--- Team/amulet_map_editor/programs/edit/api/ui/tool/base_operation_choice.py	2025-07-25 18:34:18.431106359 -0400
+++ Ryan/amulet_map_editor/programs/edit/api/ui/tool/base_operation_choice.py	2025-07-25 23:47:27.592817505 -0400
@@ -27,12 +27,13 @@
 
     ShowOpenFolder = True
 
-    def __init__(self, canvas: "EditCanvas"):
+    def __init__(self, canvas: "EditCanvas", wx_parent):
         wx.BoxSizer.__init__(self, wx.VERTICAL)
         BaseToolUI.__init__(self, canvas)
 
         self._active_operation: Optional[OperationUIType] = None
         self._last_active_operation_id: Optional[str] = None
+        self._wx_parent = wx_parent
 
         horizontal_sizer = wx.BoxSizer(wx.HORIZONTAL)
 
@@ -40,7 +41,7 @@
             self.OperationGroupName, str
         ), "OperationGroupName has not been set or is not a string."
         # The operation selection
-        self._operation_choice = SimpleChoiceAny(self.canvas)
+        self._operation_choice = SimpleChoiceAny(self._wx_parent)
         horizontal_sizer.Add(self._operation_choice)
         self._operations = UIOperationManager(self.OperationGroupName)
         self._operation_choice.SetItems(
@@ -50,7 +51,7 @@
 
         # The reload button
         self._reload_operation = wx.BitmapButton(
-            self.canvas, bitmap=image.REFRESH_ICON.bitmap(16, 16)
+            self._wx_parent, bitmap=image.REFRESH_ICON.bitmap(16, 16)
         )
         self._reload_operation.SetToolTip("Reload Operations")
         horizontal_sizer.Add(self._reload_operation)
@@ -59,7 +60,7 @@
         # The open folder button
         if self.ShowOpenFolder:
             self._open_folder = wx.BitmapButton(
-                self.canvas, bitmap=image.TABLERICONS.folder.bitmap(16, 16)
+                self._wx_parent, bitmap=image.TABLERICONS.folder.bitmap(16, 16)
             )
             self._open_folder.SetToolTip("Open Plugin Folder")
             horizontal_sizer.Add(self._open_folder)
@@ -103,7 +104,7 @@
             self._operation_sizer.Clear(delete_windows=True)
             try:
                 self._active_operation = operation(
-                    self.canvas, self.canvas, self.canvas.world
+                    self._wx_parent, self.canvas, self.canvas.world
                 )
                 self._operation_sizer.Add(
                     self._active_operation, *self._active_operation.wx_add_options
diff -ru Team/amulet_map_editor/programs/edit/api/ui/tool_manager.py Ryan/amulet_map_editor/programs/edit/api/ui/tool_manager.py
--- Team/amulet_map_editor/programs/edit/api/ui/tool_manager.py	2025-07-25 18:34:18.441106695 -0400
+++ Ryan/amulet_map_editor/programs/edit/api/ui/tool_manager.py	2025-07-25 23:47:27.602817827 -0400
@@ -1,4 +1,5 @@
 import wx
+import sys
 from typing import TYPE_CHECKING, Type, Dict, Optional
 
 from amulet_map_editor.programs.edit.api import EditCanvasContainer
@@ -32,10 +33,19 @@
         self._tools: Dict[str, BaseToolUIType] = {}
         self._active_tool: Optional[BaseToolUIType] = None
 
+        self._tool_option_panel = wx.Panel(canvas)
+        if sys.platform == "linux":
+            self._tool_option_panel.SetBackgroundColour((155, 178, 216, 255))
         self._tool_option_sizer = wx.BoxSizer(wx.VERTICAL)
+        self._tool_option_panel.SetSizer(self._tool_option_sizer)
+        self._tool_option_sizer_shove_left = wx.BoxSizer(wx.HORIZONTAL)
+        self._tool_option_sizer_shove_left.Add(self._tool_option_panel, 0, wx.EXPAND, 0)
+        self._tool_option_sizer_shove_left.AddStretchSpacer(1)
+        self.AddStretchSpacer(1)
         self.Add(
-            self._tool_option_sizer, 1, wx.EXPAND | wx.RESERVE_SPACE_EVEN_IF_HIDDEN, 0
+            self._tool_option_sizer_shove_left, 0, wx.EXPAND | wx.RESERVE_SPACE_EVEN_IF_HIDDEN, 0
         )
+        self.AddStretchSpacer(1)
 
         tool_select_sizer = wx.BoxSizer(wx.HORIZONTAL)
         tool_select_sizer.AddStretchSpacer(1)
@@ -44,12 +54,12 @@
         tool_select_sizer.AddStretchSpacer(1)
         self.Add(tool_select_sizer, 0, wx.EXPAND, 0)
 
-        self.register_tool(SelectTool)
-        self.register_tool(PasteTool)
-        self.register_tool(OperationTool)
-        self.register_tool(ImportTool)
-        self.register_tool(ExportTool)
-        self.register_tool(ChunkTool)
+        self.register_tool(SelectTool, self._tool_option_panel)
+        self.register_tool(PasteTool, self._tool_option_panel)
+        self.register_tool(OperationTool, self._tool_option_panel)
+        self.register_tool(ImportTool, self._tool_option_panel)
+        self.register_tool(ExportTool, self._tool_option_panel)
+        self.register_tool(ChunkTool, self._tool_option_panel)
 
     @property
     def tools(self):
@@ -60,11 +70,11 @@
             self._active_tool.bind_events()
         self.canvas.Bind(EVT_TOOL_CHANGE, self._enable_tool_event)
 
-    def register_tool(self, tool_cls: Type[BaseToolUIType]):
+    def register_tool(self, tool_cls: Type[BaseToolUIType], wx_parent):
         assert issubclass(tool_cls, (wx.Window, wx.Sizer)) and issubclass(
             tool_cls, BaseToolUI
         )
-        tool = tool_cls(self.canvas)
+        tool = tool_cls(self.canvas, wx_parent)
         self._tool_select.register_tool(tool.name)
         if isinstance(tool, wx.Window):
             tool.Hide()
diff -ru Team/amulet_map_editor/programs/edit/edit.py Ryan/amulet_map_editor/programs/edit/edit.py
--- Team/amulet_map_editor/programs/edit/edit.py	2025-07-25 18:34:18.441106695 -0400
+++ Ryan/amulet_map_editor/programs/edit/edit.py	2025-07-25 23:47:27.602817827 -0400
@@ -58,7 +58,7 @@
         self._temp_msg = wx.StaticText(
             self, label=lang.get("program_3d_edit.canvas.please_wait")
         )
-        self._temp_msg.SetFont(wx.Font(40, wx.DEFAULT, wx.NORMAL, wx.NORMAL))
+        self._temp_msg.SetFont(wx.Font(40, wx.DECORATIVE, wx.NORMAL, wx.NORMAL))
         self._sizer.Add(self._temp_msg, 0, flag=wx.ALIGN_CENTER_HORIZONTAL)
         self._temp_loading_bar = wx.Gauge(self, range=10000)
         self._sizer.Add(self._temp_loading_bar, 0, flag=wx.EXPAND)
diff -ru Team/amulet_map_editor/programs/edit/plugins/tools/chunk.py Ryan/amulet_map_editor/programs/edit/plugins/tools/chunk.py
--- Team/amulet_map_editor/programs/edit/plugins/tools/chunk.py	2025-07-25 18:34:18.531109726 -0400
+++ Ryan/amulet_map_editor/programs/edit/plugins/tools/chunk.py	2025-07-25 23:47:27.612818149 -0400
@@ -27,13 +27,13 @@
 
 
 class ChunkTool(wx.BoxSizer, DefaultBaseToolUI):
-    def __init__(self, canvas: "EditCanvas"):
+    def __init__(self, canvas: "EditCanvas", wx_parent):
         wx.BoxSizer.__init__(self, wx.HORIZONTAL)
         DefaultBaseToolUI.__init__(self, canvas)
 
         self._selection = ChunkSelectionBehaviour(self.canvas)
 
-        self._button_panel = wx.Panel(canvas)
+        self._button_panel = wx.Panel(wx_parent)
         button_sizer = wx.BoxSizer(wx.VERTICAL)
         self._button_panel.SetSizer(button_sizer)
 
diff -ru Team/amulet_map_editor/programs/edit/plugins/tools/import_tool.py Ryan/amulet_map_editor/programs/edit/plugins/tools/import_tool.py
--- Team/amulet_map_editor/programs/edit/plugins/tools/import_tool.py	2025-07-25 18:34:18.541110062 -0400
+++ Ryan/amulet_map_editor/programs/edit/plugins/tools/import_tool.py	2025-07-25 23:47:27.622818470 -0400
@@ -18,13 +18,13 @@
 
 
 class ImportTool(wx.BoxSizer, DefaultBaseToolUI):
-    def __init__(self, canvas: "EditCanvas"):
+    def __init__(self, canvas: "EditCanvas", wx_parent):
         wx.BoxSizer.__init__(self, wx.VERTICAL)
         DefaultBaseToolUI.__init__(self, canvas)
 
         self._selection = StaticSelectionBehaviour(self.canvas)
 
-        self._open_file_button = wx.Button(canvas, label="Import File")
+        self._open_file_button = wx.Button(wx_parent, label="Import File")
         self._open_file_button.Bind(wx.EVT_BUTTON, self._on_open_file)
         self.AddStretchSpacer()
         self.Add(self._open_file_button, flag=wx.ALL, border=10)
diff -ru Team/amulet_map_editor/programs/edit/plugins/tools/paste.py Ryan/amulet_map_editor/programs/edit/plugins/tools/paste.py
--- Team/amulet_map_editor/programs/edit/plugins/tools/paste.py	2025-07-25 18:34:18.551110399 -0400
+++ Ryan/amulet_map_editor/programs/edit/plugins/tools/paste.py	2025-07-25 23:47:27.632818792 -0400
@@ -249,7 +249,7 @@
 
 
 class PasteTool(wx.BoxSizer, DefaultBaseToolUI):
-    def __init__(self, canvas: "EditCanvas"):
+    def __init__(self, canvas: "EditCanvas", wx_parent):
         wx.BoxSizer.__init__(self, wx.HORIZONTAL)
         DefaultBaseToolUI.__init__(self, canvas)
 
@@ -258,7 +258,7 @@
         self._moving = False
         self._is_enabled = False
 
-        self._paste_panel = SimpleScrollablePanel(canvas)
+        self._paste_panel = SimpleScrollablePanel(wx_parent)
         self._paste_sizer = wx.BoxSizer(wx.VERTICAL)
         self._paste_panel.SetSizer(self._paste_sizer)
         self.Add(self._paste_panel, 0, wx.ALIGN_CENTER_VERTICAL)
diff -ru Team/amulet_map_editor/programs/edit/plugins/tools/select.py Ryan/amulet_map_editor/programs/edit/plugins/tools/select.py
--- Team/amulet_map_editor/programs/edit/plugins/tools/select.py	2025-07-25 18:34:18.551110399 -0400
+++ Ryan/amulet_map_editor/programs/edit/plugins/tools/select.py	2025-07-25 23:47:27.642819113 -0400
@@ -79,14 +79,14 @@
     _y2: wx.SpinCtrl
     _z2: wx.SpinCtrl
 
-    def __init__(self, canvas: "EditCanvas"):
+    def __init__(self, canvas: "EditCanvas", wx_parent):
         wx.BoxSizer.__init__(self, wx.HORIZONTAL)
         DefaultBaseToolUI.__init__(self, canvas)
 
         self._selection = BlockSelectionBehaviour(self.canvas)
         self._inspect_block = InspectBlockBehaviour(self.canvas, self._selection)
 
-        self._button_panel = SimpleScrollablePanel(canvas)
+        self._button_panel = SimpleScrollablePanel(wx_parent)
         button_sizer = wx.BoxSizer(wx.VERTICAL)
         self._button_panel.SetSizer(button_sizer)
 
diff -ru Team/pyproject.toml Ryan/pyproject.toml
--- Team/pyproject.toml	2025-07-25 18:34:18.571111072 -0400
+++ Ryan/pyproject.toml	2025-07-25 23:47:27.642819113 -0400
@@ -4,6 +4,6 @@
     "wheel",
     "cython >= 3.0.0a9",
     "versioneer",
-    "numpy ~= 1.17"
+    "numpy >= 1.17"
 ]
 build-backend = "setuptools.build_meta"
